control_plane/src/endpoint.rs

```diff
@@ -607,6 +613,18 @@ impl Endpoint {
                         restrict_conn: false,
                         invalid: false,
                     }]
+                }
+                // This is a hack to enable testing of the drop_subscriptions_before_start feature.
+                // Why don't we just always set the database vector properly here
+                // so that the code matched real cplane logic?
```

Comment:
For real-life logic, it should be some non-superuser role and non-postgres DB, so it's more like in the case of `create_test_user=true` <- and I'd do this, yes, let's convert this to TODO?

But I don't really understand why you need postgres DB here? It's already present and it's onwer is `cloud_admin`. It's always like that because they are created by initdb

Reply:
@lubennikovaav am I right that you need postgres DB in the list just to have perDBphase to be applied to it?

This looks very hackish. Can you just use non-superuser and non-postgres DB in test? You can probably just use the existing `create_test_user` flag when starting an endpoint (which will require some minor changes for propagating via python fixtures)